### PR TITLE
[Datasource selector] Sort datasource option list alphabetically

### PR DESCRIPTION
**Pull request by mengweieric**
### Description

Sort datasource option list in each group alphabetically.

### Issues Resolved

https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5609

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/data/public/data_sources/datasource_selector/datasource_selectable.tsx [hunk 1]
```diff
@@ -118,13 +118,24 @@ export const DataSourceSelectable = ({
     [onDataSourceSelect]
   );
 
+  const memorizedDataSourceOptionList = useMemo(() => {
+    return dataSourceOptionList.map((dsGroup: DataSourceGroup) => {
+      return {
+        ...dsGroup,
+        options: [...dsGroup.options].sort((ds1, ds2) => {
+          return ds1.label.localeCompare(ds2.label, undefined, { sensitivity: 'base' });
```
Comment (joshuali925) [inline src/plugins/data/public/data_sources/datasource_selector/datasource_selectable.tsx:126]: `base` was the original behavior in 2.9?

**Review by joshuali925 — APPROVED** [review]
is it possible to have backend to include sort in the query?

**Review by ashwin-pc — COMMENTED** [review]
This looks good to me. Think we are good to merge this once we have a unit test for this

**Comment by mengweieric** [comment]
> is it possible to have backend to include sort in the query?

It's possible but currently we are pulling datasources through different APIs and then consolidate them from front end. With this implementation we will need to compose different queries to include sorting for different datasources. For now I think to sort them after consolidations will be relatively easier.

**Review by ashwin-pc — APPROVED** [review]

**Review by ananzh — APPROVED** [review]

**Pull request merged**